metadata.tex

```diff
@@ -53,6 +54,13 @@
 \usepackage[skip=6pt, indent=0pt]{parskip}
 \setstretch{1}
 
+% Get version from env
+% \getenv{variable_name} just prints the value
+% \getenv[\macro]{variable_name} stores the value in \macro for reusability
+\newcommand{\getenv}[2][]{%
+  \CatchFileEdef{\value}{"|kpsewhich --var-value #2"}{\endlinechar=-1}%
+  \if\relax\detokenize{#1}\relax\value\else\let#1\value\fi}
```

Comment:
Why do you need this over simple `\input{.version}`?

Reply:
That's good. I didn't know about that

The `\getenv` macro might be needed for this though, unless input also reads env variables

- https://github.com/Heegu-sama/Homm3BG/actions/runs/9379614635/job/25824911861
- https://github.com/Heegu-sama/Homm3BG/pull/555/commits/122d86e576ea5e486cf35a185e47cdd5accbc71c

Reply:
Currently environment is read in perl in latexmkrc

Reply:
Can I define any variable in the latexmkrc or do I have to squueze everything into the `pre_tex_code`?

Reply:
Maybe there's an easier way (shellesc package, latexmkrc) but I was not able to make anything else work.

But I'm open to ideas :)

Reply:
I changed the `kpsewhich` command to simple `echo` to get the env variable. That should be more acceptable :)

Reply:
A controversial question: will it work on Windows?

Reply:
Does it have to? I mean all scripts are written in bash :)

Reply:
The owner doesn't work actively on this any more, but up until now you could just open the project in TeXStudio and build over there. I'm afraid that with this change it will become impossible.

Reply:
OMG, it breaks at getenv, not here haha.

Reply:
Anyway, I'll follow your recommendations anyway on this topic but supporting Windows is just another burden to bear. So unless there's a good reason to do so (like at least few active devs), I would discourage that

Reply:
So if I go back to the previous command, I still don't have a way to test that it works properly. I could assume it'll work but still